classes/controller/FrontController.php

```diff
@@ -1746,8 +1746,6 @@ protected function updateQueryString(array $extraParams = null)
             }
         }
 
-        ksort($params);
```

Comment:
ok but... why this function was here at first place? I'm afraid someone in 1 year will re-introduce it for the same reason, don't you think?

Reply:
I don't understand why it was here at the first place. Our canonical URLs are not generated with their parameters sorted alphabetically.

If we want to regenerate an URL and comparing it with the default canonical URL, we have to reuse them in the same order as they are in the current address.